### PR TITLE
Orders management

### PR DESCRIPTION
**Pull request by Anies12**
### Related:
related to #37
Orders management page, Fron-end

### CODE REVIEW

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+    error: "",
+    filter: false
   };
 
   componentDidMount() {
     axios
       .get("/api/v1//viewOrders")
       .then(res => {
-        this.setState({ orders: res.data });
+        if (res.status === 204) {
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: no need to check if there is content or not, the empty table doesn't make any problem

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 2]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+    });
+    return filtered;
+  }else{
+    return object;
+  }
 }
+  };
+
+  filter = async (type, value) => {
+    console.log(type, value)
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete console log

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 3]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+    const { date, status, name, orders } = this.state;
+    let filtered = [];
+    if (type === "date") {
+      await this.setState({ date: value });
+      if (status) {
+        filtered = this.statusFilter(orders);
+        console.log(222555,filtered)
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete console log

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 4]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+        if (name) {
+          filtered = this.nameFilter(filtered);
+        }
+        filtered = this.dateFilter(filtered);
+        console.log(222666,filtered)
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete console log

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 5]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+        this.setState({ filteredOrders: filtered, filter: true });
+      } else if (name) {
+        filtered = this.nameFilter(orders);
+        filtered = this.dateFilter(filtered);
+        this.setState({ filteredOrders: filtered, filter: true });
+      } else if (value.length > 0) {
+        filtered = this.dateFilter(orders);
+        console.log(filtered)
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete console log

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 6]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+    } else if (type === "status") {
+      await this.setState({ status: value });
+      if (date) {
+        console.log(777)
+        filtered = this.dateFilter(orders);
+        if (name) {
+          filtered = this.nameFilter(filtered);
+        }
+        filtered = this.statusFilter(filtered);
+        console.log(66666,filtered)
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete console log

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 7]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+        filtered = this.dateFilter(orders);
+        console.log(filtered)
+        this.setState({ filteredOrders: filtered, filter: true });
+      } else {
+        this.setState({ filteredOrders: [], filter: false });
+      }
+    } else if (type === "status") {
+      await this.setState({ status: value });
+      if (date) {
+        console.log(777)
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete console log

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 8]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+                <Button
+                  className="ordersManagement_addOrder-button"
+                  type="primary"
+                >
+                  إضافة طلب
+                </Button>
+                <div className="ordersManagement_filters-container">
+                  <div className="ordersManagement_filters-container-timeFilter">
+                    <p
+                      style={{ textAlign: "right" }}
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete inline style

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 9]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+                      ? this.state.filteredOrders
+                      : this.state.orders
+                  }
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    } else if (this.state.error.response.status === 204) {
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: no need for this if statement, the empty table no make any sense, but handle the loading data from the back end.

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 10]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+                  <Button
+                    className="ordersManagement_addOrder-button"
+                    type="primary"
+                  >
+                    إضافة طلب
+                  </Button>
+                  <div className="ordersManagement_filters-container">
+                    <div className="ordersManagement_filters-container-timeFilter">
+                      <p
+                        style={{ textAlign: "right" }}
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: inline style

File: server/controller/index.js [hunk 11]
```diff
@@ -1,12 +1,16 @@
...
 const express = require('express');
+const orders = require('./order/index');
 
 const { check } = require('./middleware/authentication');
 const { protect } = require('./middleware/protectRoute');
 
 const router = express.Router();
 
+
+router.get('/viewOrders', orders.get);
```
Comment (hshahwan) [inline server/controller/index.js]: add this reoute in index of oders folder

File: server/controller/order/index.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
+const getOrdersQuery = require('../../database/queries/order/getOrders');
+
+exports.get = (req, res) => {
+  let orders = [];
+
+  //   getOrdersQuery()
```
Comment (hshahwan) [inline server/controller/order/index.js]: add the query to select data from fake data

File: server/controller/order/index.js [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+  // .then((resposne) => {
+  //   if (!Response[0]) {
+  //   res.status(204).send('No orders yet');
+  //   } else {
+  // res.send(JSON.stringify(resposne));
+  //   }
+  // })
+  // .catch(error => res.status(500).send(JSON.stringify({ error })));
+
+  orders = JSON.stringify([
```
Comment (hshahwan) [inline server/controller/order/index.js]: delete object fake data, and add it into fakedata.sql

File: server/controller/order/index.js [hunk 14]
```diff
@@ -0,0 +1,115 @@
+const getOrdersQuery = require('../../database/queries/order/getOrders');
+
+exports.get = (req, res) => {
+  let orders = [];
+
+  //   getOrdersQuery()
+  // .then((resposne) => {
+  //   if (!Response[0]) {
+  //   res.status(204).send('No orders yet');
```
Comment (hshahwan) [inline server/controller/order/index.js]: delete comment code

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 15]
```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
...
+                  >
+                    إفراغ الحقول
+                  </Button>
+                </div>
+                <TableComponent
+                  pageName="orders"
+                //   viewPopup={viewPopup}
+                //   editPopup={editPopup}
+                  deletePopup={deletePopup}
+                  editHtml="<input type='text' placeholder='name' id='name'>"
```
Comment (hshahwan) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: delete editHtml, we don't need any more

Comment (hshahwan) [reply client/src/Components/Layouts/Ordersmanagement/index.jsx]: and update the parameters in all tables.

File: client/src/App.jsx [hunk 16]
```diff
@@ -8,10 +8,12 @@ import Navbar from "./Components/CommonComponent/Navbar";
 import Profile from "./Components/Layouts/SingleCustomer";
 
 import Login from "./Components/Layouts/Login";
+import Ordersmanagement from "./Components/Layouts/Ordersmanagement/index";
```
Comment (tomduggan85) [inline client/src/App.jsx]: @Anies12 This looks like a duplicate import (with a slightly different name) so you can probably delete this line.

File: client/src/App.jsx [hunk 17]
```diff
@@ -29,10 +31,15 @@ class App extends Component {
               exact
               path="/captains/profile/:id"
               render={() => (
-                <div>
+                <div className="app" >
```
Comment (tomduggan85) [inline client/src/App.jsx]: @Anies12 It might be better to use a different className than "app" here, because that className is already being used up above.

File: client/src/Components/CommonComponent/Table/Table.jsx [hunk 18]
```diff
@@ -71,7 +70,9 @@ class TableCmponent extends Component {
             dataSource={columns}
             pagination={{
               pageSize: isNaN(this.state.pageSize)
-                ? columns.length
+                ? columns
+                  ? columns.length
```
Comment (tomduggan85) [inline client/src/Components/CommonComponent/Table/Table.jsx]: @Anies12 This is a little hard to follow, when is it supposed to use `columns.length`?

File: client/src/App.jsx (hunk 17 above)
Comment (Anies12) [reply client/src/App.jsx]: thanks Tom, this file is actually being edited with every pull request from each one of us, that's why you can see a mess here :)

File: client/src/Components/CommonComponent/Table/style.css [hunk 19]
```diff
@@ -1,14 +1,50 @@
+input[type="text"]::placeholder {
+  text-align: right;
+}
+input[type="text"]:-ms-input-placeholder {
+  text-align: right;
+}
```
Comment (tomduggan85) [inline client/src/Components/CommonComponent/Table/style.css]: @Anies12 These styles are global, so if you want to ensure that they only apply to the Table component you'll need to make them a little more specific, for example `.table-container input[type="text"]::placeholder {`

File: client/src/Components/Layouts/Customers/addcustomer.jsx [hunk 20]
```diff
@@ -46,8 +46,8 @@ const CollectionCreateForm = Form.create({ name: 'form_in_modal' })(
           onCancel={onCancel}
           onOk={onCreate}
           className="modalcontainer"
-        >
-          <Form className='modalform'>
+        >modalform
```
Comment (tomduggan85) [inline client/src/Components/Layouts/Customers/addcustomer.jsx]: @Anies12 This looks like it might have been accidentally moved by your IDE, did you mean to move this "modalform" from being a classname to being plain text?

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 21]
```diff
@@ -1,60 +1,284 @@
...
+                  .setHours(0, 0, 0, 0) <= toDate
+              ) {
+                return true;
+              }
+            });
+            return filtered;
+          }
+        }
+      } else {
+        return object;
```
Comment (tomduggan85) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: @Anies12 there is an if-statement that doesn't have an else (the one on line 60) so you could probably change this last block, 79-81, to just return (and not be an else statement) and that would handle all of the else conditions.

File: client/src/Components/Layouts/Ordersmanagement/index.jsx [hunk 22]
```diff
@@ -1,60 +1,284 @@
...
+      }
+    }
+  };
+
+  filter = async (type, value) => {
+    const { date, status, name, orders } = this.state;
+    let filtered = [];
+    if (type === "date") {
```
Comment (tomduggan85) [inline client/src/Components/Layouts/Ordersmanagement/index.jsx]: @Anies12 I think this big function might be easier to read and manage if you break it up into three functions (filterDate, filterStatus, filterName) instead of having the big if-statement for `type`

File: client/src/Components/CommonComponent/Table/Table.jsx (hunk 18 above)
Comment (Anies12) [reply client/src/Components/CommonComponent/Table/Table.jsx]: if the pageSize is a number and there exist columns

File: client/src/App.jsx (hunk 17 above)
Comment (tomduggan85) [reply client/src/App.jsx]: 😂 I think every App.js ever written has that same problem

File: client/src/Components/CommonComponent/Table/style.css (hunk 19 above)
Comment (Anies12) [reply client/src/Components/CommonComponent/Table/style.css]: well, I'm actually doing that is many other places, but I wanted to ensure that every input field in the application is aligned to the right, as we are building an Arabic-language application

File: client/src/Components/CommonComponent/Table/Table.jsx (hunk 18 above)
Comment (tomduggan85) [reply client/src/Components/CommonComponent/Table/Table.jsx]: maybe `isNaN(this.state.pageSize) && columns ? columns.length : parseInt(this.state.pageSize)` would work?

File: client/src/Components/CommonComponent/Table/style.css (hunk 19 above)
Comment (tomduggan85) [reply client/src/Components/CommonComponent/Table/style.css]: Ah got it.  In that case, maybe these styles belong in a more "global" location, like App.css or something.  But it's not super important 🤷‍♂️

File: client/src/Components/Layouts/Customers/addcustomer.jsx (hunk 20 above)
Comment (Anies12) [reply client/src/Components/Layouts/Customers/addcustomer.jsx]: Oh that's right :) , thank u

File: client/src/Components/Layouts/Ordersmanagement/index.jsx (hunk 22 above)
Comment (Anies12) [reply client/src/Components/Layouts/Ordersmanagement/index.jsx]: you are right, I realized that after I have done this and was already struggling to keep going

Comment (Anies12) [reply client/src/Components/Layouts/Ordersmanagement/index.jsx]: you are right, I realized that after I did this and I was struggling to keep going :)

Comment (tomduggan85) [reply client/src/Components/Layouts/Ordersmanagement/index.jsx]: haha yeah, it sometimes helps to come back to it the next day after getting some sleep, to get a fresh perspective.

File: client/src/Components/Layouts/Ordersmanagement/index.jsx (hunk 21 above)
Comment (Anies12) [reply client/src/Components/Layouts/Ordersmanagement/index.jsx]: yes it would, thank u

File: client/src/Components/Layouts/Ordersmanagement/index.jsx (hunk 22 above)
Comment (Anies12) [reply client/src/Components/Layouts/Ordersmanagement/index.jsx]: haha, this is the only thing that I could do

File: client/src/Components/CommonComponent/Table/style.css (hunk 19 above)
Comment (Anies12) [reply client/src/Components/CommonComponent/Table/style.css]: sure, it would be better to put it in App.css

File: client/src/Components/CommonComponent/Table/Table.jsx (hunk 18 above)
Comment (Anies12) [reply client/src/Components/CommonComponent/Table/Table.jsx]: yeah, it would :) , this is a lot easier to read

Comment (tomduggan85) [reply client/src/Components/CommonComponent/Table/Table.jsx]: Cool.  Whenever I see or write a nested ternary like that, I try to find a way to refactor it.
